src/webhook/pkg/mutating_webhook/mutating-webhook_test.go

```diff
@@ -0,0 +1,568 @@
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("1"),
+							corev1.ResourceMemory: resource.MustParse("1Gi"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("1"),
+							corev1.ResourceMemory: resource.MustParse("1Gi"),
+						},
+					},
+				},
+				{
+					// GPU container
+					Name: "container3",
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:                     resource.MustParse("1"),
+							corev1.ResourceName(GPU_RESOURCE_NAME): resource.MustParse("1"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:                     resource.MustParse("1"),
+							corev1.ResourceName(GPU_RESOURCE_NAME): resource.MustParse("1"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podMutator := &PodMutator{}
+	mutatedPod, err := podMutator.mutatePod(pod)
+	if err != nil {
+		t.Fatalf("mutatePod() error = %v", err)
+	}
+
+	// Check annotations
+	if mutatedPod.Annotations[PEERPODS_CPU_ANNOTATION] != "6" {
+		t.Errorf("Expected CPU annotation to be 6, got %s", mutatedPod.Annotations[PEERPODS_CPU_ANNOTATION])
+	}
+
+	if mutatedPod.Annotations[PEERPODS_MEMORY_ANNOTATION] != "8192" {
+		t.Errorf("Expected Memory annotation to be 8192, got %s", mutatedPod.Annotations[PEERPODS_MEMORY_ANNOTATION])
+	}
+
+	if mutatedPod.Annotations[PEERPODS_GPU_ANNOTATION] != "1" {
+		t.Errorf("Expected GPU annotation to be 1, got %s", mutatedPod.Annotations[PEERPODS_GPU_ANNOTATION])
+	}
+
+	// Check resource requirements for all containers are cleared
+	// Except for the first container which has the peer-pod resource added
+	for idx, container := range mutatedPod.Spec.Containers {
+		// Skip the first container
+		if idx == 0 {
+			continue
+		}
+
+		if len(container.Resources.Requests) != 0 || len(container.Resources.Limits) != 0 {
+			t.Errorf("Expected resources to be cleared for container %d, got Requests: %v, Limits: %v",
+				idx, container.Resources.Requests, container.Resources.Limits)
+		}
+	}
+
+	// Check resource requirements for all init containers are cleared
+	for idx, container := range mutatedPod.Spec.InitContainers {
+		if len(container.Resources.Requests) != 0 || len(container.Resources.Limits) != 0 {
+			t.Errorf("Expected resources to be cleared for init container %d, got Requests: %v, Limits: %v",
+				idx, container.Resources.Requests, container.Resources.Limits)
+		}
+	}
+
+	// Check that the first container has the peer-pod resource added
+	expectedResource := resource.MustParse("1")
+	if !mutatedPod.Spec.Containers[0].Resources.Requests[corev1.ResourceName(POD_VM_EXTENDED_RESOURCE_DEFAULT)].Equal(expectedResource) {
+		t.Errorf("Expected VM resource request to be 1, got %v",
+			mutatedPod.Spec.Containers[0].Resources.Requests[corev1.ResourceName(POD_VM_EXTENDED_RESOURCE_DEFAULT)])
+	}
+}
+
+func TestMutatePod_Negative(t *testing.T) {
```

Comment:
Optional suggestion: Maybe the name `TestMutatePod_NoChangeForDifferentRuntimeClass` would be easier to understand at a glance in test output?